vignettes/articles/multi-scale-resampling.Rmd

```diff
@@ -0,0 +1,233 @@
+
+plot_measurements <- readr::read_csv("https://www.mm218.dev/junkdrawer/riemann/plot_measurements.csv") |> 
+  st_as_sf(coords = c("long", "lat"), crs = 4326) |> 
+  st_transform(5070)
+```
+
+And this is what we're going to go ahead and resample. We want to assess our model's performance at multiple scales, following the approach in Riemann et al. That means we need to do the following:
+
+1. Block our study area using multiple sets of regular hexagons of different sizes, and assign our data to the hexagon it falls into within each set.
+2. Perform leave-one-block-out cross-validation with each of those sets, fitting our model to n - 1 of the n hexagons we've created and assessing it on the hold-out hexagon.
+3. Calculate model accuracy for each size based on those held-out hexes.
+
+So to get started, we need to block our study area. We can do this using the `spatial_block_cv()` function from spatialsample. We'll generate ten different sets of hexagon tiles, using `cellsize` arguments of between 10,000 and 100,000 meters[^3]. The code to do that, and to store all of our resamples in a single tibble, looks like this[^4]:
+
+[^3]: This value is in meters because our coordinate reference system is in meters. It represents the length of the [apothem](https://en.wikipedia.org/wiki/Apothem), from the center of each polygon to the middle of the side. We're using hexagons because Riemann et al. also used hexagons. 
+
+[^4]: `v` is `Inf` because we want to perform leave-one-block-out cross-validation, but we don't know how many blocks there will be before they're created. This is the supported way to do leave-one-X-out cross-validation in spatialsample > 0.2.0 (another option is to set `v = NULL`).
+
+```{r, message=FALSE}
+set.seed(123)
+library(dplyr)
+library(spatialsample)
+cellsize <- seq(10, 100, 10) * 1000
+
+riemann_resamples <- tibble(
+  resamples = purrr::map(
+    cellsize, 
+    \(cs) {
+      spatial_block_cv(
+        plot_measurements,
+        v = Inf,
+        cellsize = cs,
+        square = FALSE
+      )
+    }
+  ),
+  cellsize = cellsize
+)
+```
+
+If we want, we can visualize a few of our resamples, to get a sense of what our tiling looks like:
+
+```{r}
+autoplot(riemann_resamples$resamples[[9]])
+```
+
+```{r}
+autoplot(riemann_resamples$resamples[[10]])
+```
+
+And that's step 1 of the process completed! Now we need to move on to step 2, and actually fit models to each of these resamples. I want to highlight that this is a _lot_ of models, and so is going to take a while[^5]:
+
+[^5]: Linear regression was invented in 1805, ish, long before the Analytical Engine was a twinkle in Babbage's eye. Whenever I get frustrated at how long fitting multiple models like this takes, I like to take a step back and recognize that I am asking my poor overworked computer to fit roughly as many models as were used in the first ~100 years of the technique's life. 
+
+```{r}
+purrr::map_dbl(
+  riemann_resamples$resamples,
+  nrow
+) |> 
```

Comment:
I haven't said this yet, but let's use the magrittr pipe for now, since we do have plenty of users who are on slightly older versions of R.